src/runtime/entitlements-manager.js

```diff
@@ -390,15 +404,18 @@ export class EntitlementsManager {
   }
 
   /**
+   * @param {?string} opt_encryptedDocumentKey
    * @return {!Promise<!Entitlements>}
    * @private
    */
-  fetch_() {
-    const url = serviceUrl(
-        '/publication/' +
-        encodeURIComponent(this.publicationId_) +
-        '/entitlements');
-    return this.fetcher_.fetchCredentialedJson(url)
+  fetch_(opt_encryptedDocumentKey) {
+    let url = '/publication/' +
+    encodeURIComponent(this.publicationId_) +
+    '/entitlements';
+    if (opt_encryptedDocumentKey) {
+      url = url + '/crypt/' + opt_encryptedDocumentKey;
```

Comment:
Hmm. I don't see why it should be in the path - this is a non-identifying parameter. It should be a query param (and later maybe even a POST param).

Another critical issue: this parameter must be encoded for URL.